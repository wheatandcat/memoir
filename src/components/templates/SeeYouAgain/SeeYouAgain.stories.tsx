import React from 'react';
import SeeYouAgain, { Props } from './SeeYouAgain';

const props = (): Props => ({});

export default {
  title: 'templates/SeeYouAgain',
};

export const _SeeYouAgain = () => <SeeYouAgain {...props()} />;

_SeeYouAgain.story = {
  name: 'SeeYouAgain',
};
