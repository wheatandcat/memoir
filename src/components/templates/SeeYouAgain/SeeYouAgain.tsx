import React, { memo } from 'react';
import { StyleSheet, ScrollView } from 'react-native';
import View from 'components/atoms/View';
import Text from 'components/atoms/Text';
import theme from 'config/theme';
import { SafeAreaView } from 'react-native-safe-area-context';
import Button from 'components/atoms/Button';
import Image from 'components/atoms/Image';
import { useSetRecoilState } from 'recoil';
import { screenState } from 'store/atoms';

export type Props = {};

const SeeYouAgain: React.FC<Props> = () => {
  const setScreenState = useSetRecoilState(screenState);

  return (
    <SafeAreaView>
      <View style={styles.root}>
        <ScrollView>
          <View style={styles.inner}>
            <View pt={4} pb={5}>
              <Image
                source={require('../../../img/icon/trust.png')}
                width={100}
                height={100}
                contentFit="contain"
              />
            </View>
            <Text size="base">
              アカウント削除が完了しました。{'\n\n'}
              ご利用ありがとうございました。{'\n\n'}また、会いましょう。
            </Text>
            <View py={5}>
              <Button
                title="TOPに戻る"
                onPress={() => {
                  setScreenState({ seeYouAgain: false });
                }}
                width={200}
              />
            </View>
          </View>
        </ScrollView>
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  root: {
    backgroundColor: theme().color.background.light,
    height: '100%',
  },
  inner: {
    alignItems: 'center',
    justifyContent: 'center',
    width: '100%',
    padding: theme().space(2),
    marginBottom: theme().space(6),
  },
});

export default memo(SeeYouAgain);
