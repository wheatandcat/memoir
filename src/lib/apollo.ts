import { setContext } from '@apollo/client/link/context';
import { Alert } from 'react-native';
import { onError } from '@apollo/client/link/error';
import { ApolloClient, InMemoryCache, createHttpLink } from '@apollo/client';
import { storageKey, getItem } from 'lib/storage';
import Auth from 'lib/auth';

export const cache = new InMemoryCache();
const auth = new Auth();

type Param = {
  Authorization?: string;
  UserID?: string;
};

const makeApolloClient = async () => {
  const uri = `${process.env.API_HOST}/query`;

  const authLink = setContext(async (_, { headers }) => {
    const h = headers;
    const param: Param = {};
    const token = await auth.getIdToken();

    if (token) {
      param.Authorization = `Bearer ${token}`;
    } else {
      const userID = await getItem(storageKey.USER_ID_KEY);

      if (userID) {
        param.UserID = userID;
      }
    }

    return {
      headers: {
        ...h,
        ...param,
      },
    };
  });

  const errorLink = onError((error) => {
    console.log(error.graphQLErrors);
    if ((error.graphQLErrors || []).length > 0) {
      const graphQLErrors = error.graphQLErrors || [
        { message: 'エラー発生しました' },
      ];

      const message = graphQLErrors[0].message;
      Alert.alert('エラー', message);
    }
  });

  return new ApolloClient({
    link: errorLink.concat(authLink.concat(createHttpLink({ uri }))),
    cache: cache,
  });
};

export default makeApolloClient;
