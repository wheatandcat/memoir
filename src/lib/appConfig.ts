import firebase from 'lib/system/firebase';

export type AppConfig = {
  maintenance: boolean;
  maintenanceMessage: string;
  maintenancePeriod: string | null;
  supportVersion: string;
};

export const defaultAppConfig = () => ({
  maintenance: false,
  maintenanceMessage: '',
  maintenancePeriod: null,
  supportVersion: '1.0.0',
});

export const getAppConfig = async (
  db: firebase.firestore.Firestore
): Promise<AppConfig> => {
  const qs = await db.collection('appConfig').limit(1).get();

  const records = qs.docs.map((elem) => {
    return elem.data();
  });

  if (!records || !records[0]) {
    return defaultAppConfig();
  }

  return records[0] as AppConfig;
};
