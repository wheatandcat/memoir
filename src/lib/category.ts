import { StyleSheet, ViewStyle } from 'react-native';
import theme from 'config/theme';
import master from 'lib/master';

export const categoryBorderStyle = (category: number): ViewStyle => {
  switch (category) {
    case master.CATEGORY_1:
      return styles.category1;
    case master.CATEGORY_2:
      return styles.category2;
    case master.CATEGORY_3:
      return styles.category3;
  }

  return styles.nonCategory;
};

const styles = StyleSheet.create({
  nonCategory: {
    borderLeftColor: theme().color.background.main,
  },
  category1: {
    borderLeftColor: theme().category.color.category1,
  },
  category2: {
    borderLeftColor: theme().category.color.category4,
  },
  category3: {
    borderLeftColor: theme().category.color.category2,
  },
});
