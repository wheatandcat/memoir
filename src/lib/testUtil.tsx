import React from 'react';
import { GraphQLHandler, GraphQLRequest } from 'msw';
import {
  ApolloClient,
  ApolloProvider,
  createHttpLink,
  InMemoryCache,
} from '@apollo/client';
import { render } from '@testing-library/react-native';
import { server } from 'mocks/server';
import fetch from 'cross-fetch';

const link = createHttpLink({
  uri: 'http://localhost:8080/query',
  fetch,
  credentials: 'same-origin',
});

const client = new ApolloClient({
  link,
  uri: 'http://localhost:8080/query',
  cache: new InMemoryCache(),
});

export const testRenderer =
  (children: React.ReactNode) =>
  (responseOverride?: GraphQLHandler<GraphQLRequest<never>>) => {
    if (responseOverride) {
      server.use(responseOverride);
    }
    return render(<ApolloProvider client={client}>{children}</ApolloProvider>);
  };
