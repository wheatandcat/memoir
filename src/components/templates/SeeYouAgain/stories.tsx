import React from 'react';
import { storiesOf } from '@storybook/react-native';
import SeeYouAgain, { Props } from './SeeYouAgain';

const props = (): Props => ({});

storiesOf('templates/SeeYouAgain', module).add('SeeYouAgain', () => (
  <SeeYouAgain {...props()} />
));
