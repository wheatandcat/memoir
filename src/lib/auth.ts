import firebase from 'lib/system/firebase';
import AsyncStorage from '@react-native-async-storage/async-storage';
import { storageKey } from 'lib/storage';
import dayjs from 'lib/dayjs';

const initFirebaseAuth = (): Promise<firebase.User | null> => {
  return new Promise((resolve) => {
    var unsubscribe = firebase.auth().onAuthStateChanged((user) => {
      // user オブジェクトを resolve
      resolve(user);

      // 登録解除
      unsubscribe();
    });
  });
};

class Auth {
  setSession = async (refresh = false) => {
    const user = await initFirebaseAuth();
    if (!user) {
      return null;
    }

    await AsyncStorage.setItem(storageKey.AUTH_UID_KEY, user.uid);

    const result = await user.getIdTokenResult(refresh);

    await AsyncStorage.setItem(storageKey.AUTH_ID_TOKEN_KEY, result.token);
    await AsyncStorage.setItem(
      storageKey.AUTH_ID_TOKEN_EXPIRATION_KEY,
      String(result.claims.exp)
    );

    return result.token;
  };
  getIdToken = async () => {
    const idToken = await AsyncStorage.getItem(storageKey.AUTH_ID_TOKEN_KEY);
    if (!idToken) {
      return null;
    }

    const expiration = await AsyncStorage.getItem(
      storageKey.AUTH_ID_TOKEN_EXPIRATION_KEY
    );

    if (Number(expiration) > dayjs().unix()) {
      return idToken;
    }

    return this.setSession(true);
  };
  logout = async () => {
    await firebase.auth().signOut();

    await AsyncStorage.removeItem(storageKey.AUTH_UID_KEY);
    await AsyncStorage.removeItem(storageKey.AUTH_ID_TOKEN_KEY);
    await AsyncStorage.removeItem(storageKey.AUTH_ID_TOKEN_EXPIRATION_KEY);
  };
}

export default Auth;
