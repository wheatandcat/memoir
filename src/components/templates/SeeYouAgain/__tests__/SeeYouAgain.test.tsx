import React from 'react';
import { testRenderer } from 'lib/testUtil';
import { screen } from '@testing-library/react-native';
import SeeYouAgain, { Props } from '../SeeYouAgain';

const propsData = (): Props => ({});

describe('components/templates/SeeYouAgain/SeeYouAgain.tsx', () => {
  it('正常にrenderすること', () => {
    testRenderer(<SeeYouAgain {...propsData()} />)();
    expect(screen.findAllByText('TOPに戻る')).toBeTruthy();
  });
});
