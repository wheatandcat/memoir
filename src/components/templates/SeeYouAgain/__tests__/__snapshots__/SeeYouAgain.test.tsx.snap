// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/templates/SeeYouAgain/SeeYouAgain.tsx 正常にrenderすること 1`] = `
<ForwardRef>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "height": "100%",
      }
    }
  >
    <ScrollView>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginBottom": 128,
            "padding": 8,
            "width": "100%",
          }
        }
      >
        <View
          pb={5}
          pt={4}
        >
          <Image
            height={100}
            resizeMode="contain"
            source={null}
            width={100}
          />
        </View>
        <Text
          fontFamily="RobotoCondensed-Bold"
          size="base"
          variants="body"
        >
          アカウント削除が完了しました。
          


          ご利用ありがとうございました。
          


          また、会いましょう。
        </Text>
        <View
          py={5}
        >
          <Button
            fontFamily="NotoSansJP-Bold"
            onPress={[Function]}
            size="base"
            title="TOPに戻る"
            width={200}
          />
        </View>
      </View>
    </ScrollView>
  </View>
</ForwardRef>
`;
